parser/Builder.cc

```diff
@@ -1206,6 +1206,10 @@ class Builder::Impl {
                 // Label key is a symbol `sym: val`
                 return match_var_hash(label->loc, key->val.show(gs_));
             } else if (auto *key = parser::cast_node<DSymbol>(pair->key.get())) {
+                if (key->nodes.empty()) {
+                    error_without_recovery(ruby_parser::dclass::InvalidKey, key->loc);
```

Comment:
Can we recover to a tree like this?

```
{:"" => <ErrorNode>}
```

?

It seems like we're already in the semantic actions so recovering with an error node seems like it would be easy enough.

Reply:
Yep, we could return an `ErrorNode` instead of an `""` name.